docs/source/caching/http-cache.mdx

```diff
@@ -2,81 +2,49 @@
+// Configure a max size of 100MB
+val maxSize = 100 * 1024 * 1024
 
-// Size in bytes of the cache
-val maxSize: Long = 1024 * 1024
-
-// Build the ApolloClient
-val apolloClient = ApolloClient.Builder()
-    .serverUrl("/")
-    .httpCache(directory = directory, maxSize = maxSize)
-    .build()
-                   
-// Control the fetch policy  
-val query = ApolloRequest.Builder(FeedQuery(limit = 10, type = FeedType.HOT))
-    .httpFetchPolicy(HttpFetchPolicy.CacheFirst)
+apolloClient = ApolloClient.Builder()
+    .networkTransport(
+        HttpNetworkTransport(
+            httpRequestComposer = DefaultHttpRequestComposer(mockServer.url()),
+            engine = CachingHttpEngine(dir, maxSize)
+        )
+    )
     .build()
-val dataResponse = apolloClient.query(query)
-```
-
-```java:title=Java
-// Directory where cached responses will be stored
-File directory = new File(cacheDir, "apolloCache");
-
-// Size in bytes of the cache
-long maxSize = 1024 * 1024;
-
-// Build the ApolloClient
-ApolloClient apolloClient = HttpCacheExtensionsKt.httpCache(
-    new ApolloClient.Builder().serverUrl("/"),
-    directory,
-    maxSize
-  )
-  .build();
-Rx2ApolloClient rx2ApolloClient = new Rx2ApolloClient(apolloClient, Schedulers.io());
-
-// Control the fetch policy
-ApolloRequest<FeedQuery.Data> query = HttpCacheExtensionsKt.httpFetchPolicy(
-    new ApolloRequest.Builder<>(new FeedQuery(10, FeedType.HOT)),
-    HttpFetchPolicy.CacheFirst
-  )
-  .build();
-
-ApolloResponse<FeedQuery.Data> response = rx2ApolloClient.query(query).blockingGet();
 ```
 
-</MultiCodeBlock>
+The HTTP cache is a least recently used (LRU) cache with a configurable max size.
 
-**IMPORTANT:** Caching is provided only for `query` operations. It isn't available for `mutation` operations.
+Once your cache setup is complete, the cache will be used by default by all your queries. By default, queries will try to find a result in the cache first and go the network if it's not there. This is the `HttpFetchPolicy.CacheFirst`. You can customize that behaviour with `httpFetchPolicy(HttpFetchPolicy)`:
 
-There are four available cache policies in `HttpFetchPolicy`:
+```kotlin
+val response = apolloClient.query(query)
+                 // Don't use the cache
+                 .httpFetchPolicy(HttpFetchPolicy.NetworkOnly)
 
-- `NetworkOnly` - Fetch a response from the network only, ignoring any cached responses. This is the default.
-- `CacheOnly` - Fetch a response from the cache only, ignoring the network. If the cached response doesn't exist or is expired, then return
-  an error.
-- `CacheFirst` - Fetch a response from the cache first. If the response doesn't exist or is expired, then fetch a response from the network.
-- `NetworkFirst` - Fetch a response from the network first. If the network fails and the cached response isn't expired, then return cached
-  data instead.
+                 // Or only use the cache
+                 .httpFetchPolicy(HttpFetchPolicy.CacheOnly)
```

Comment:
Maybe a note saying that in this case `HttpCacheMissException` will be thrown in case of a cache miss?